ranger-zookeeper/src/main/java/com/flipkart/ranger/zookeeper/servicefinderhub/ZkServiceDataSource.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2015 Flipkart Internet Pvt. Ltd.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.flipkart.ranger.zookeeper.servicefinderhub;
+
+import com.flipkart.ranger.core.finderhub.ServiceDataSource;
+import com.flipkart.ranger.core.model.Service;
+import com.google.common.base.Preconditions;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.curator.framework.CuratorFrameworkFactory;
+import org.apache.curator.retry.ExponentialBackoffRetry;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ *
+ */
+@Slf4j
+public class ZkServiceDataSource implements ServiceDataSource {
+
+    private final String namespace;
+    private String connectionString;
+    private CuratorFramework curatorFramework;
+    private boolean curatorProvided;
+
+    public ZkServiceDataSource(String namespace,
+                               String connectionString,
+                               CuratorFramework curatorFramework){
+        this.namespace = namespace;
+        this.connectionString = connectionString;
+        this.curatorFramework = curatorFramework;
+    }
+
+    @Override
+    public Collection<Service> services() throws Exception {
+        final List<String> children = curatorFramework.getChildren()
```

Comment:
`val`. Also there is no need to create this variable. There are no checks etc in the middle. What happens if no child exists? does it reutn null or empty list? if it's empty list, you can skip this vairable and return by chaining the lambda directly here and return.  If null, you need to handle that and return empty in that case

Reply:
There is no need to create that variable. Collapsed it.